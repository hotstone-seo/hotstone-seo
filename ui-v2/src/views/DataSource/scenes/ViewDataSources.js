import React from 'react';
import PropTypes from 'prop-types';
import { Link, useHistory } from 'react-router-dom';
import { PageHeader, Button, message } from 'antd';
import { deleteDataSource } from 'api/datasource';
import useDataSources from 'hooks/useDataSources';
import { DataSourceList } from 'components/DataSource';

import { PlusCircleOutlined } from '@ant-design/icons';

function ViewDataSources({ match }) {
  const [dataSources, setDataSources] = useDataSources();
  const history = useHistory();

  const showEditScene = (dataSource) => {
    history.push(`${match.url}/${dataSource.id}`);
  };

  const removeDataSource = (dataSource) => {
    deleteDataSource(dataSource.id)
      .then(() => {
        message.success(`Successfully deleted ${dataSource.name}`);
        setDataSources(dataSources.filter((item) => item.id !== dataSource.id));
      })
      .catch((error) => {
        message.error(error.message);
      });
  };
  const addDataSource = () => {
    history.push(`${match.url}/new`);
  };
  return (
    <div>
      <PageHeader
        title="Data Sources"
        subTitle="Manage location for retrieving resources"
        style={{ background: '#fff' }}
      />
      <div style={{ padding: 24 }}>
        <Button type="primary" style={{ marginBottom: 16 }} icon={<PlusCircleOutlined />}  onClick={() => addDataSource()}>
           Add New Data Source
        </Button>
        <DataSourceList
          dataSources={dataSources}
          onClick={showEditScene}
          onEdit={showEditScene}
          onDelete={removeDataSource}
        />
      </div>
    </div>
  );
}

ViewDataSources.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string,
  }).isRequired,
};

export default ViewDataSources;
