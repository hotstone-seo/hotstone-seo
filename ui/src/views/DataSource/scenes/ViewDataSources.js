import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import { PageHeader, Button, message } from 'antd';
import { fetchDataSources, deleteDataSource } from 'api/datasource';
import useAsync from 'hooks/useAsync';
import { DataSourceList } from 'components/DataSource';

import { PlusOutlined } from '@ant-design/icons';

function ViewDataSources({ match }) {
  const history = useHistory();
  const {
    pending, value: dataSources, setValue: setDataSources, error,
  } = useAsync(fetchDataSources);

  if (error) {
    message.error(error.message);
  }

  const showEditScene = (dataSource) => {
    history.push(`${match.url}/${dataSource.id}`);
  };

  const removeDataSource = (dataSource) => {
    deleteDataSource(dataSource.id)
      .then(() => {
        message.success(`Successfully deleted ${dataSource.name}`);
        setDataSources(
          dataSources.filter((item) => item.id !== dataSource.id),
        );
      })
      .catch((err) => {
        message.error(err.message);
      });
  };

  const addDataSource = () => {
    history.push(`${match.url}/new`);
  };

  return (
    <div>
      <PageHeader
        title="Data Sources"
        subTitle="Manage location for retrieving resources"
        style={{ background: '#fff' }}
      />
      <div style={{ padding: 24 }}>
        <Button
          type="primary"
          style={{ marginBottom: 16 }}
          icon={<PlusOutlined />}
          onClick={() => addDataSource()}
        >
          Add New Data Source
        </Button>
        <DataSourceList
          dataSources={dataSources}
          loading={pending}
          onClick={showEditScene}
          onEdit={showEditScene}
          onDelete={removeDataSource}
        />
      </div>
    </div>
  );
}

ViewDataSources.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string,
  }).isRequired,
};

export default ViewDataSources;
