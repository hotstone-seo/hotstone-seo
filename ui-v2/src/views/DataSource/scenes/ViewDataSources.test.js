import React from 'react';
import {
  render, wait,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import mockAxios from 'jest-mock-axios';

import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ViewDataSources from './ViewDataSources';

configure({ adapter: new Adapter() });

const respMock = [{
  id: 1, name: 'FooDS', url: '/foo-ds', updated_at: new Date().toISOString(), created_at: new Date().toISOString(),
}];

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

test('View Mismatch rule unit test', () => {
  const props = {
    match: { url: 'tes.com' },
  };
  const tree = mount(<ViewDataSources {...props} />);
  expect(tree).toMatchSnapshot();
});
/*
describe('ViewDataSources', () => {
  test('first load', async () => {
    const url = '/datasources';
    const {
      queryByText,
    } = render(<ViewDataSources match={{ url }} />);

    await wait(() => {
      expect(mockAxios.get).toHaveBeenCalledWith('/data_sources');
      mockAxios.mockResponse({ data: respMock });

      expect(queryByText(/Data Sources/)).toBeInTheDocument();
      expect(queryByText(/FooDS/)).toBeInTheDocument();
    });
  });
});
*/