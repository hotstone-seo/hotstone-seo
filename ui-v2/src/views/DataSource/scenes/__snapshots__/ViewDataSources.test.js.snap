// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Mismatch rule unit test 1`] = `ReactWrapper {}`;
